src/main/java/org/apache/mesos/scheduler/DefaultScheduler.java

```diff
@@ -124,6 +129,31 @@ private void initializeRecoveryScheduler() {
                 recoveryStatusRef);
     }
 
+    private void initializeApi() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                JettyApiServer apiServer = new JettyApiServer(
```

Comment:
This is limiting. What if the framework/service author wants to add more resources, other than {Plan, Recovery, State}Resource ?